app/memory_service.py

```diff
@@ -0,0 +1,26 @@
+from typing import Dict, Any
+from datetime import datetime, timezone
+from uuid import UUID, uuid4
+
+
+class MemoryService:
+    def __init__(self):
+        self.memories = {}
+
+    def create_memory(self, user_id, content):
+        memory_id = uuid4()
+        memory_data = {
```

Comment:
You can introduce a model for MemoryData which will be used in the storage layer (potentially as DB model in the future). Don't use free-form dict/JSON.